zkvm/src/predicate.rs

```diff
@@ -198,43 +201,60 @@ mod tests {
         let op = pred.prove_program_predicate(&bytecode(&prog2), &[]);
         assert!(op.verify().is_err());
     }
-
+    
     #[test]
-    fn valid_disjunction() {
+    fn valid_disjunction1() {
         let gens = PedersenGens::default();
 
         // dummy predicates
-        let left = Predicate::Opaque(gens.B.compress());
-        let right = Predicate::Opaque(gens.B_blinding.compress());
-
-        let pred = left.clone().or(right.clone()).unwrap();
-        let op = pred.prove_or(&left, &right);
+        let mut preds: Vec<Predicate> = Vec::with_capacity(2);
+        let first = Predicate::Opaque(gens.B.compress());
+        let second = Predicate::Opaque(gens.B_blinding.compress());
```

Comment:
No need to create a Vec and then push twice, if you can use `vec!` macro:
```suggestion
        let preds = vec![Predicate::Opaque(gens.B.compress()), Predicate::Opaque(gens.B_blinding.compress())];
```